File: leetcode/Javascript Solutions/findDuplicate.js
let findDuplicate = (nums)=>{
    nums.sort((a,b)=>a-b)
    for(let i=0;i<nums.length-1;i++){
        if(nums[i]==nums[i+1]){
            return nums[i]
        }
    }
    return 0
}

File: problem-solving-swift/leetcode/findDuplicate.swift
//
//  findDuplicate.swift
//  problem-solving-swift
//
//  Created by Avijeet on 09/04/24.
//

import Foundation

class FindDuplicate {
    static func find(_ nums: [Int]) -> Int {
        var map : [Int: Int] = [:]
        
        var repeating = -1
        
        for element in nums {
            if let count = map[element] {
                map[element] = count+1
            } else {
                map[element] = 1
            }
        }
        
        map.forEach { (key, value) in
            if value > 1 {
                repeating = key
            }
        }
        
        return repeating
    }
}
